tests/foreman/cli/test_errata.py

```diff
@@ -349,27 +364,37 @@ def cv_publish_promote(cv, org, lce):
     :param lce: lifecycle environment
     :type lce: entities.LifecycleEnvironment
     """
-    ContentView.publish({'id': cv.id})
-    cvv = ContentView.info({'id': cv.id})['versions'][-1]
-    ContentView.version_promote(
+    sat.cli.ContentView.publish(
         {
-            'id': cvv['id'],
-            'organization-id': org.id,
+            'id': cv.id,
+            'organization': org,
+            'lifecycle-environment-ids': lce.id,
+        }
+    )
+    cv = cv.read()
+    cvv = cv.version[-1].read()
+    sat.cli.ContentView.version_promote(
+        {
+            'id': cvv.id,
+            'organization': org,
+            'content-view-id': cv.id,
             'to-lifecycle-environment-id': lce.id,
+            'force': force,
         }
     )
 
 
-def cv_filter_cleanup(filter_id, cv, org, lce):
+def cv_filter_cleanup(sat, filter_id, cv, org, lce):
     """Delete the cv filter, then publish and promote an unfiltered version."""
-    ContentViewFilter.delete(
+    sat.cli.ContentViewFilter.delete(
         {
             'content-view-id': cv.id,
             'id': filter_id,
             'organization-id': org.id,
         }
     )
-    cv_publish_promote(cv, org, lce)
+    cv = cv.read()
+    cv_publish_promote(sat, cv, org, lce, force=True)
```

Comment:
Locally in test teardown I received errors for promoting to next env out of sequence, use force to bypass. 
Only on `cv_filter_cleanup` I have it forced, could it be doing something out of order?

Reply:
Looks like it is passing, i think the assertion fail here was related to the `module_ak_cv_lce` fixture that I'm no longer using.

Reply:
Oh, maybe I know. Is the function called twice with the same CV in some test (or its teardown)?

Reply:
Yes, we call it once in this test at [L-818](https://github.com/SatelliteQE/robottelo/blob/4e38a466e9bd08ab90e368399e23e323bc0216e6/tests/foreman/cli/test_errata.py#L818)
`cv_publish_promote(target_sat, module_cv, module_entitlement_manifest_org, module_lce)`

Then we call once more during teardown (`def cleanup(): cv_filter_cleanup()` at [L-800](https://github.com/SatelliteQE/robottelo/blob/4e38a466e9bd08ab90e368399e23e323bc0216e6/tests/foreman/cli/test_errata.py#L800)) , 
to publish&promote a version without the `cv_filter` once it has been deleted.

`cv_filter_cleanup` calls the same `cv_publish_promote` that the test did prior.

Reply:
for cleanup instead of publishing/promoting the new version without `cv_filter`,
could instead just call `target_sat.cli.ContentView.remove_version` for the ver. created in the test ?